zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1965,6 +1968,8 @@ public <T> T retryUntilConnected(final Callable<T> callable)
     if (_monitor != null) {
       _monitor.increaseOutstandingRequestGauge();
     }
+    int retryCount = 1;
+    ExponentialBackoffStrategy retryStrategy = new ExponentialBackoffStrategy(_operationRetryTimeoutInMillis, true);
```

Comment:
Thanks for addressing the comment! 
Since `_operationRetryTimeoutInMillis` is set at client level, do you think we can make this object as an object variable?

Reply:
We are combining cleanup and actual fix in the same review. Let me split the review in 2 different PR. one purely cleanup and another using the new cleaned-up API.